typer/_completion_shared.py

```diff
@@ -100,13 +100,13 @@ def install_bash(*, prog_name: str, complete_var: str, shell: str) -> Path:
     # It seems bash-completion is the official completion system for bash:
     # Ref: https://www.gnu.org/software/bash/manual/html_node/A-Programmable-Completion-Example.html
     # But installing in the locations from the docs doesn't seem to have effect
-    completion_path = Path.home() / f".bash_completions/{prog_name}.sh"
+    completion_path = Path.home() / ".bash_completions" / f"{prog_name}.sh"
     rc_path = Path.home() / ".bashrc"
     rc_path.parent.mkdir(parents=True, exist_ok=True)
     rc_content = ""
     if rc_path.is_file():
         rc_content = rc_path.read_text()
-    completion_init_lines = [f"source {completion_path}"]
+    completion_init_lines = [f"source '{completion_path}'"]
```

Comment:
It's safe to quote in posix, too.
In fact it's the right thing to do, even on posix, because the home directory might contain spaces

However, I recommend quoting using `shlex.quote`, which will also convert other corner cases

Reply:
`shlex.quote` isn't actually supposed to be used on Windows, it's marked in the documentation as being unsafe and unreliable: https://docs.python.org/3/library/shlex.html#shlex.quote.

Reply:
I'd very much prefer to avoid using `shlex.quote` altogether. I'd propose to merge this PR as is, and see whether any further issues get reported that would warrant a more complex solution.